Sources/MapboxCoreNavigation/Directions+RoutingProvider.swift

```diff
@@ -8,30 +8,30 @@ extension URLSessionDataTask: NavigationProviderRequest {
 }
 
 extension Directions: RoutingProvider {
-    
+
     @discardableResult public func calculateRoutes(options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> NavigationProviderRequest? {
-        return calculate(options, completionHandler: completionHandler)
```

Comment:
Source of the offline route not functioning. If we instead call calculateWithCache, the Nav UI should also be able to leverage offline routing functionality.

Reply:
Did you try to use `MapboxRoutingProvider` instead? It is meant to be offline-ready, while Directions is online-only provider. I would suggest not to change these lines since it effectively makes `Directions` object a wrapper for `MapboxRoutingProvider` which is implicit and not desired.

Reply:
The issue isn't out app's code base. The issue is that NavSDK isn't leveraging an offline friendly approach.

Are you suggesting changing the method call here? Or somewhere else?

Reply:
Hi @Udumft  ,

I’m looking to address navigation being unable to reroute if offline.

While this approach may not be ideal, if you have a preferred approach i would be happy following it.

The problem I'm running into is that we’ve advertised offline navigation and although it works in other parts of our app, **specifically during navigation** it's failing due to not using a method capable of falling back to offline calculations

Reply:
MapboxRoutingProvider does function, I just wasn't sure how best to force the NavSDK to use it. Given your slack suggestion, I went with the injected NavigationService which allows configuration of a RoutingProvider.